framework/test_cases/cloud_run_testcase.py

```diff
@@ -0,0 +1,167 @@
+        cls.compute_v1 = ComputeV1(
+            cls.gcp_api_manager, cls.project, version=cls.compute_api_version
+        )
+        cls.region = xds_flags.CLOUD_RUN_REGION.value
+
+    def initTrafficDirectorManager(self) -> TrafficDirectorManager:
+        return TrafficDirectorCloudRunManager(
+            self.gcp_api_manager,
+            project=self.project,
+            resource_prefix=self.resource_prefix,
+            resource_suffix=self.resource_suffix,
+            network=self.network,
+            compute_api_version=self.compute_api_version,
+            enable_dualstack=self.enable_dualstack,
+        )
+
+    def initKubernetesClientRunner(self, **kwargs) -> KubernetesClientRunner:
+        return KubernetesClientRunner(
+            k8s.KubernetesNamespace(
+                self.k8s_api_manager, self.client_namespace
+            ),
+            deployment_name=self.client_name,
+            image_name=self.client_image,
+            td_bootstrap_image=self.td_bootstrap_image,
+            gcp_project=self.project,
+            gcp_api_manager=self.gcp_api_manager,
+            gcp_service_account=self.gcp_service_account,
+            xds_server_uri=self.xds_server_uri,
+            network=self.network,
+            debug_use_port_forwarding=self.debug_use_port_forwarding,
+            enable_workload_identity=self.enable_workload_identity,
+            deployment_template="client.deployment.yaml",
+            stats_port=self.client_port,
+            reuse_namespace=self.server_namespace == self.client_namespace,
+            **kwargs,
+        )
+
+    def setupServerBackends(
+        self,
+        *,
+        wait_for_healthy_status=True,
+        server_runner=None,
+        max_rate_per_endpoint: Optional[int] = None,
+    ):
+        if server_runner is None:
+            server_runner = self.server_runner
+        service_url = server_runner.get_service_url()
+        self.td.backend_service_add_backends(
+            [service_url], max_rate_per_endpoint=max_rate_per_endpoint
+        )
+        if wait_for_healthy_status:
+            self.td.wait_for_backends_healthy_status(
+                replica_count=server_runner.replica_count
+            )
+
+    def startTestServers(self, server_runner=None) -> List[XdsTestServer]:
+        if server_runner is None:
+            self.server_runner = CloudRunServerRunner(
+                project=self.project,
+                service_name=self.server_namespace,
+                image_name=self.server_image,
+                network=self.network,
+                region=self.region,
+                gcp_api_manager=self.gcp_api_manager,
+            )
+        test_servers = self.server_runner.run()
+        for test_server in test_servers:
+            test_server.set_xds_address(
+                self.server_xds_host, self.server_xds_port
+            )
+        return test_servers
+
+    def assertEDSConfigExists(self, config):
```

Comment:
Add `@override` to make it clear we're replacing parent's assertion method. Unfortunately, we have too many of them to track.